ppa/archive/views.py

```diff
@@ -247,7 +253,56 @@ class DigitizedWorkDetailView(DetailView):
     model = DigitizedWork
     slug_field = 'source_id'
     slug_url_kwarg = 'source_id'
+    paginate_by = 50
 
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        # pull in the query if it exists to use
+        query = self.request.GET.get('query', '')
+
+        solr_pageq = None
+        if query:
+            context['query'] = query
+            digwork = context['object']
+            solr_q = 'text:(%s) AND srcid:("%s") AND item_type:(page)' \
+                % (query, digwork.source_id)
+            solr_opts = {
+                'q': solr_q,
+                'sort': 'order asc',
+                'fl': '*',
+                'hl': True,
+                'hl.fl': 'content',
+                # No way to set this to actual number in one search
+                # therefore using an arbitrarily high but not system breaking
+                # number.
+                'hl.snippets': 10000,
```

Comment:
This number is how many snippets to return for each item that matches, so it really doesn't need to be big.  Probably only 2-3 ?